### PR TITLE
Fix some resource leaks during installation

### PR DESCRIPTION
**Pull request by JetXujing**
when rpm install src rpm, some resource is leaked, fix it.

### CODE REVIEW

**Comment by JetXujing** [comment]
@pmatilai PTAL

**Comment by pmatilai** [comment]
> @pmatilai PTAL

All PR's will get looked at, in time. Please avoid these kind of personal pings unless it's actually personal, ie there's something that requires a response from that person specifically.

**Comment by JetXujing** [comment]
> > @pmatilai PTAL
> 
> All PR's will get looked at, in time. Please avoid these kind of personal pings unless it's actually personal, ie there's something that requires a response from that person specifically.

I'm sorry. I'll follow this rule in the future.

**Comment by ffesti** [comment]
You can't reference some local files in the commit message ([fix file leak when install src rpm which is URL](https://github.com/rpm-software-management/rpm/pull/2289/commits/3938cefacedbfd1ae40ff356c9d8d43e173da902)). You need to include the content if it is appropriate. Or describe the issue.

File: lib/header.c [hunk 1]
```diff
@@ -241,6 +241,8 @@ Header headerFree(Header h)
 	}
 	h->index = _free(h->index);
     }
+    if (h->blob)
```
Comment (pmatilai) [inline lib/header.c:244]: Don't bother checking for non-NULL, free() does it anyway.

**Comment by JetXujing** [comment]
> You can't reference some local files in the commit message ([fix file leak when install src rpm which is URL](https://github.com/rpm-software-management/rpm/pull/2289/commits/3938cefacedbfd1ae40ff356c9d8d43e173da902)). You need to include the content if it is appropriate. Or describe the issue.

Thank you for your comments. I've made changes.

File: lib/header.c (hunk 1 above)
Comment (JetXujing) [reply lib/header.c]: Thank you for your comments. I've made changes.

**Comment by pmatilai** [comment]
Merged manually to tweak the commit messages a little bit, primarily to capitalize sentences.

Thanks a lot for the patches! While we do check for memory leaks every now and then, clearly nobody has done so for src.rpm installations in a long, long time if ever.

**Pull request closed without merging**